sfxclient/gometrics.go

```diff
@@ -12,14 +12,34 @@ var startTime = time.Now()
 // GoMetricsSource is a singleton Collector that collects basic go system stats.  It currently
 // collects from runtime.ReadMemStats and adds a few extra metrics like uptime of the process
 // and other runtime package functions.
-var GoMetricsSource Collector = &goMetrics{}
+var GoMetricsSource MetricSource = &goMetrics{}
 
-type goMetrics struct{}
+// MetricSource interface implements the necessary api's for goMetric
+type MetricSource interface {
+	Collector
+	AddCallback(func(stats runtime.MemStats))
```

Comment:
Since this is memstats, let's make this interface called MemCallback or something letting us add other stats at some point